Symfony/CS/Fixer/Symfony/MethodSeparationFixer.php

```diff
@@ -156,6 +156,38 @@ private function fixSpaceAboveMethod(Tokens $tokens, $classStart, $methodIndex)
             }
         }
 
+        // deal with comments above a method
+        if ($tokens[$nonWhiteAbove]->isGivenKind(T_COMMENT)) {
+            if ($firstMethodAttrIndex - $nonWhiteAbove === 1) {
+                // no white space found between comment and method start
+                $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstMethodAttrIndex, 1);
+            } else {
+                // $tokens[$nonWhiteAbove+1] is always a white space token here
+                $lineBreakCount = substr_count($tokens[$nonWhiteAbove + 1]->getContent(), "\n");
+                if ($lineBreakCount > 1) {
+                    // more than one line break, always bring it back to 2 line breaks between the method start and what is above it
+                    $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstMethodAttrIndex, 2);
+                } else {
+                    // there are 2 cases:
+                    if ($tokens[$nonWhiteAbove - 1]->isWhitespace() && substr_count($tokens[$nonWhiteAbove - 1]->getContent(), "\n")) {
+                        // 1. the comment is meant for the method (although not a PHPDoc)
+                        //      make sure there is one line break between the method and the comment
+                        $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstMethodAttrIndex, 1);
+                        //      make sure the is blank line above the comment (with the exception when it is directly after a class opening)
+                        $nonWhiteAboveComment = $tokens->getNonWhitespaceSibling($nonWhiteAbove, -1);
+                        $this->correctLineBreaks($tokens, $nonWhiteAboveComment, $nonWhiteAbove, $nonWhiteAboveComment === $classStart ? 1 : 2);
+                    } else {
+                        // 2. the comment belongs to the code above the method
+                        //      make sure there is a blank line above the method (i.e. 2 line breaks)
+                        $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstMethodAttrIndex, 2);
+                    }
+                }
+            }
```

Comment:
The `if{if-else{if-else{if-else}}}`-construct is hard to read ;)

I would prefer some more `return`s instead of the `else`s.